src/mdns/mdns_mdnssd.cpp

```diff
@@ -703,6 +877,290 @@ void Publisher::Destroy(Publisher *aPublisher)
+    }
+}
+
+void PublisherMDnsSd::ServiceSubscription::GetAddrInfo(uint32_t aInterfaceIndex)
+{
+    assert(mServiceRef == nullptr);
+
+    otbrLog(OTBR_LOG_INFO, "[mdns] DNSServiceGetAddrInfo %s inf %d", mInstanceInfo.mHostName.c_str(), aInterfaceIndex);
+
+    DNSServiceGetAddrInfo(&mServiceRef, /* flags */ 0, aInterfaceIndex,
+                          kDNSServiceProtocol_IPv6 | kDNSServiceProtocol_IPv4, mInstanceInfo.mHostName.c_str(),
+                          HandleGetAddrInfoResult, this);
+}
+
+void PublisherMDnsSd::ServiceSubscription::HandleGetAddrInfoResult(DNSServiceRef          aServiceRef,
+                                                                   DNSServiceFlags        aFlags,
+                                                                   uint32_t               aInterfaceIndex,
+                                                                   DNSServiceErrorType    aErrorCode,
+                                                                   const char *           aHostName,
+                                                                   const struct sockaddr *aAddress,
+                                                                   uint32_t               aTtl,
+                                                                   void *                 aContext)
+{
+    static_cast<ServiceSubscription *>(aContext)->HandleGetAddrInfoResult(aServiceRef, aFlags, aInterfaceIndex,
+                                                                          aErrorCode, aHostName, aAddress, aTtl);
+}
+
+void PublisherMDnsSd::ServiceSubscription::HandleGetAddrInfoResult(DNSServiceRef          aServiceRef,
+                                                                   DNSServiceFlags        aFlags,
+                                                                   uint32_t               aInterfaceIndex,
+                                                                   DNSServiceErrorType    aErrorCode,
+                                                                   const char *           aHostName,
+                                                                   const struct sockaddr *aAddress,
+                                                                   uint32_t               aTtl)
+{
+    OTBR_UNUSED_VARIABLE(aServiceRef);
+    OTBR_UNUSED_VARIABLE(aInterfaceIndex);
+
+    Ip6Address address;
+
+    otbrLog(OTBR_LOG_INFO, "DNSServiceGetAddrInfo reply: %d, flags=%u, host=%s, sa_family=%d", aErrorCode, aFlags,
+            aHostName, aAddress->sa_family);
+
+    VerifyOrExit(aErrorCode == kDNSServiceErr_NoError);
+    VerifyOrExit((aFlags & kDNSServiceFlagsAdd) && aAddress->sa_family == AF_INET6);
+
+    address.CopyFrom(*reinterpret_cast<const struct sockaddr_in6 *>(aAddress));
+    VerifyOrExit(!address.IsLinkLocal(),
+                 otbrLog(OTBR_LOG_DEBUG, "[mdns] DNSServiceGetAddrInfo ignore link-local address %s",
+                         address.ToString().c_str()));
```

Comment:
What will happen in this case? Will it result in no address or will we get more IPv6 addresses when the mDNSResponder finds more?

Reply:
it will wait for more IPv6 addresses. The process stop when `kDNSServiceFlagsMoreComing` flag is 1.